### ISSUE TITLE
Connecting with bluewallet on lndhub using testnet

### ISSUE THREAD

**Issue by ohenrik** [BlueWallet/LndHub#22]
Hi

I have set up a lndhub running on testnet. I think I can connect correctly to lndhub and create lightning wallets, however i do not understand how i can fill my lightning wallet with funds. It looks like BlueWallet by default creates mainnet wallets. So they are not compatible with testnet.

Are there any other way to refilling my lightning wallets than using the internal wallets in the app?

**Comment by Overtorment** [comment]
Youll have to get a refill address for your LN wallet and send some testnet coins with a wallet of your choice.

**Comment by ohenrik** [comment]
Thank you for answering :) So my next question is: How do i get the refill address?

I am not able to find my lightning wallet address in the app.

**Comment by Overtorment** [comment]
Create LN wallet in BW -> open it -> tap 'manage funds' -> tap 'refill' ->
choose source wallet -> refill screen should open, with destination address
pre-filled in input field

**Comment by ohenrik** [comment]
Thank you, I'll have decided to temporarely just working with a small amount on mainnet. I'll consider coming back to this later and possibly contribute to bluewallet to make this simpler.

**Comment by inspiraluna** [comment]
When I tap on "manage funds" on my empty lightning wallet. Refill appears which I tap also: 
Then my Bitcoin Wallet with some coins appears. As I choose that wallet, I receive:
"API error: bad auth (code 1)"

in the log of my LndHub Server I can see :
info: [/getbtc] :["bla bla bla "]
bla bla bla ::x:y.z.a.b - - [15/Mar/2019:09:29:09 +0000] "GET /getbtc HTTP/1.1" 304 - "-" "okhttp/3.12.1"

and after a while:
updateLightning()
updated

Is it possible to give a better logging on what could be wrong here? Or do you have any other idea what could be the case here?

Thanks in advance!

**Comment by Overtorment** [comment]
Something is wrong with the lndhub account, it cant authorize. Maybe you
are connecting to wrong lndhub uri..?

Also, testnet is not supported

**Comment by inspiraluna** [comment]
I should open a new issue. Since I don't use testnet.
Where can I configure the accounts on LndHub?

**Comment by inspiraluna** [comment]
you mean lndhub cannot authenticate correctly on LightningHub? or Bitcoind?

**Comment by Overtorment** [comment]
When you create ln wallet in Bluewallet it creates credentials in lndhub
(stored in redis)

**Comment by inspiraluna** [comment]
Thanks! Seemed like redis didn't work. So now had to delete the unused wallet and created a new one. I could transfer funds. But can't see yet my Satoshis. Might have to wait 6 blocks... ?!

**Comment by Overtorment** [comment]
3 blocks. Also, try pull-to-refresh

**Comment by ianthius** [comment]
How do you get LNDHub to run on a testnet instance of LND and bitcoind? Is there an option in the config file? I am looking through the code trying to figure it out since it's not mentioned anywhere. I am getting an error when launching LNDHub:
**bitcoind failure**

that I think is related to it not being able to RPC properly since it woudn't be including the -testnet flag

**Comment by Overtorment** [comment]
@ianthius testnet is not supported.  `bitcoind failure` suggests that bitcoind is not configured correctly

**Comment by ianthius** [comment]
I suspect that the bitcoind failure is due to the fact that bitcoind is running a testnet node, not a mainnet node.

So LNDHub was developed on mainnet only? Do you have a few hints at what I should be looking at changing / adding to get testnet running on LNDHub? The very first comment in this issue mentions that he had a testnet LNDHub up and running, you are saying this is not possible?

I'd like to be able to run a testnet LNDHub to test this well prior to using it in production.

**Comment by kiwiidb** [comment]
Sorry for the necromancy. Is testnet supported in the mean time? If not, what are the main things that are needed for it to be?

I'm running dockerized lndhub with a testnet lnd backend. I've tried to fill up the wallet using "refill from an external wallet" from Bluewallet, which gives you a testnet address. But after more than 3 confirmations, no funds show up.

I've seen some things in the Redis database:
- the addresses I found under the `listtransactions` key have a `bc1` prefix, not a `tb1`.
- there are some transactions in the `listtransactions` value that I don't recognize at all.
